source/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator.cc

```diff
@@ -439,6 +444,28 @@ std::string SPIFFEValidator::extractTrustDomain(const std::string& san) {
   return "";
 }
 
+void SPIFFEValidator::initializeCertExpirationStats(Stats::Scope& scope,
+                                                    const std::string& cert_name) {
+  // TODO(peterl328): Due to current interface, we only receive one cert name.
+  // Since we may have multiple certificates here, we will use the provided cert name and append
+  // an index to it. Assumes the order in the ca_certs_ vector doesn't change.
+  int idx = 0;
+  for (bssl::UniquePtr<X509>& cert : spiffe_data_->ca_certs_) {
+    // Add underscore between cert name and index to avoid collisions
+    std::string cert_name_with_idx = absl::StrCat(cert_name, "_", idx);
+
+    Stats::Gauge*& expiration_gauge = expiration_gauges_map_[cert_name_with_idx];
```

Comment:
This map is never populated, and isn't needed. You don't need to keep a pointer to the gauge; just call `createCertificateExpirationGauge`, set the value, then you're done.